app/models/pull_request.rb

```diff
@@ -186,20 +200,17 @@ def validate_mergeable
     end
   end
 
-  def validate_status
+  def validate_status(saved_changes)
     label = Label.tests_fail
-    # get current status
-    state = begin
-              statuses = Github.client.combined_status(gh_repository_id, github[:head][:sha])
-              statuses[:state]
-            rescue StandardError => e
-              Raven.capture_message('validate status', extra: { error: e.inspect })
-              nil
-            end
-
-    case state
+
+    case status
     when 'failure'
+      # if CI failed, add a label to repo
       repository.ensure_label_exists(label)
+      # if CI failed AND was green, add a new comment
+      # TODO: Check if it was `success` previously / Check what it was before it was `pending`
```

Comment:
do we want to add a comment if status changes from failure -> pending -> failure ? Probably not.